src/serializer/ResidualFunctions.js

```diff
@@ -225,7 +225,12 @@ export class ResidualFunctions {
     for (let [funcBody, instances] of functionEntries) {
       let functionInfo = this.residualFunctionInfos.get(funcBody);
       invariant(functionInfo);
-      this._referentialize(functionInfo.unbound, instances);
+      let additionalFunctionInstances = instances.filter((instance) => rewrittenAdditionalFunctions.has(instance.functionValue))
+      if (additionalFunctionInstances && additionalFunctionInstances.length) {
+        invariant(additionalFunctionInstances.length === instances.length, additionalFunctionInstances.length);
+      } else {
+        this._referentialize(functionInfo.unbound, instances);
```

Comment:
`_referentialize` deals with bindings that are (potentially) mutated by residual functions.
How do we deal with this conceptually when the same bindings are used by prepacked residual functions?